cortex-m-rt/macros/src/lib.rs

```diff
@@ -328,7 +328,7 @@ pub fn exception(args: TokenStream, input: TokenStream) -> TokenStream {
             f.block.stmts = iter::once(
                 syn::parse2(quote! {{
                     // check that this exception actually exists
-                    exception::#ident;
+                    cortex_m_rt::exception::#ident;
```

Comment:
We should use `::cortex_m_rt` here. However, we are still on the 2015 edition here, right ? So we also have to add `extern crate cortex_m_rt`, are we inside a macro created module/function here ? Because if not, the `extern crate` might collide with another one inserted by the user.

Reply:
Hmm, good point. We could bump to 2018? Our MSRV is already well beyond when 2018 came out.

Reply:
Oh, the macros crate is actually 2018, is it okay to emit invalid 2015 code from a 2018 proc-macros crate ?

Reply:
Incredibly it seems "yes": https://doc.rust-lang.org/edition-guide/editions/advanced-migrations.html#macro-hygiene

> Macros use a system called "edition hygiene" where the tokens within a macro are marked with which edition they come from. This allows external macros to be called from crates of varying editions without needing to worry about which edition it is called from.

Nevertheless I'd be in favour of bumping to 2018 edition for c-m-rt too...

Reply:
Agreed.